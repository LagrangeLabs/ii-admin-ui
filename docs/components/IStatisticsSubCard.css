.statisticssubcard-container {
  display: flex;
  flex-wrap: wrap;
}
.ii-ui-statisticssubcard {
  min-width: auto;
  background: #e5f7ec;
  width: calc((100% - 40px) / 3);
}
.ii-ui-statisticssubcard + .ii-ui-statisticssubcard {
  margin-left: 20px;
}
.statisticssubcard-title {
  color: rgb(70, 52, 231);
}
.statisticssubcard-icon {
  color: #a5e3f0;
  font-size: 14px;
}
.green-bg {
  background:rgba(147, 229, 147, 0.3);
}
.red-bg {
  background: rgba(224, 116, 116, 0.2);
}
.statisticssubcard-icon {
  color: #1ab857;
  background: rgba(26, 184, 87, 0.15);
}
.statisticssubcard-value {
  color: #1ab857;
}
.statisticssubcard-unit {
  color: blue;
}
