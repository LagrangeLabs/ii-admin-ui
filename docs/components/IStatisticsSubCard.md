---
title: IStatisticsSubCard组件
---

# IStatisticsSubCard 组件

统计卡片：主要用于展示当前页面的所属纬度的可视化数据信息

Demo:

```tsx
import React from 'react';
import { IStatisticsSubCard } from 'ii-admin-ui';
import { InfoCircleOutlined, QuestionCircleOutlined } from '@ant-design/icons';

import './IStatisticsSubCard.css';

export default () => {
  return (
    <div className="statisticssubcard-container">
      <IStatisticsSubCard
        className="green-bg"
        title="外呼进度"
        tips="这是外呼进度卡片"
      ></IStatisticsSubCard>
      <IStatisticsSubCard
        className="statisticssubcard-item red-bg"
        title="外呼进度"
        titleClass="statisticssubcard-title"
        icon={
          <InfoCircleOutlined className="ii-ui-statisticssubcard-header-icon" />
        }
        tips="这是外呼进度卡片"
        value={0}
        unit="w"
      ></IStatisticsSubCard>
      <IStatisticsSubCard
        className="statisticssubcard-item red-bg"
        title="外呼进度"
        titleClass="statisticssubcard-title"
        icon={<InfoCircleOutlined className="statisticssubcard-icon" />}
        tips="这是外呼进度卡片"
        value="40"
        valueClass="statisticssubcard-value"
        unit="w"
        unitClass="statisticssubcard-unit"
      ></IStatisticsSubCard>
    </div>
  );
};
```

<!-- <API src='../../src/components/IStatisticsSubCard'> -->

### IStatisticsSubCard prop 说明

| 属性       | 说明                     | 类型             | 默认值                      | 是否必传 | 版本 |
| ---------- | ------------------------ | ---------------- | --------------------------- | -------- | ---- |
| className  | 设置卡片的样式名         | string           |                             | 否       |      |
| title      | 卡片名称                 | string           |                             | 是       |      |
| titleClass | 卡片名称样式名           | string           |                             | 否       |      |
| icon       | 图标 icon 组件           | React.ReactNode  | `<QuestionCircleOutlined/>` | 否       |      |
| tips       | 图标 icon 划过的展示信息 | string           |                             | 否       |      |
| value      | 卡片值                   | string \| number | --                          | 否       |      |
| valueClass | 卡片值样式名             | string           |                             | 否       |      |
| unit       | 卡片值的单位             | string           | %                           | 否       |      |
| unitClass  | 卡片值的单位样式         | string           |                             | 否       |      |
