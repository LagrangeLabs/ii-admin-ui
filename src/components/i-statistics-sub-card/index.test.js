import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import IStatisticsSubCard from './index';

describe("Test IStatisticsSubCard component on the self's props", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should have the props class', () => {
    const wrapper = mount(
      <IStatisticsSubCard
        className="statisticssubcard-item red-bg"
        title="外呼进度"
        titleClass="statisticssubcard-title"
        icon={null}
        tips="这是外呼进度卡片"
        value={0}
        valueClass="statisticssubcard-value"
        unit="%"
        unitClass="statisticssubcard-unit"
      ></IStatisticsSubCard>,
    );
    expect(
      wrapper.find('.ii-ui-statisticssubcard.statisticssubcard-item.red-bg')
        .length,
    ).toBeTruthy(); // 判断有没有对应的className
    expect(
      wrapper.find(
        '.ii-ui-statisticssubcard-header-title.statisticssubcard-title',
      ).length,
    ).toBeTruthy(); // 判断有没有对应的titleClass
    expect(
      wrapper.find(
        '.ii-ui-statisticssubcard-content-value.statisticssubcard-value',
      ).length,
    ).toBeTruthy(); // 判断有没有对应的valueClass
    expect(
      wrapper.find(
        '.ii-ui-statisticssubcard-content-unit.statisticssubcard-unit',
      ).length,
    ).toBeTruthy(); // 判断有没有对应的unitClass

    expect(wrapper.text()).toMatch(/外呼进度/); // 判断有没有对应的title
    // expect(wrapper.text()).toMatch(/这是外呼进度卡片/); // 无法判断有没有对应的tips，因为只有在hover到icon的时候才显示
    expect(wrapper.text()).toMatch(/0/); // 判断有没有对应的value
    expect(wrapper.text()).toMatch(/%/); // 判断有没有对应的unit

    expect(wrapper.render()).toMatchSnapshot();
    wrapper.unmount();
  });
});
