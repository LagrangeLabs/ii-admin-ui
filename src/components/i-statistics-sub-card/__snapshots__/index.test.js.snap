// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test IStatisticsSubCard component on the self's props should have the props class 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "ii-ui-statisticssubcard statisticssubcard-item red-bg",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ii-ui-statisticssubcard-header",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "ii-ui-statisticssubcard-header-title statisticssubcard-title",
            },
            "children": Array [
              Node {
                "data": "外呼进度",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h2",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {},
              "children": Array [],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {},
            "children": Array [],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "ii-ui-statisticssubcard-header-title statisticssubcard-title",
              },
              "children": Array [
                Node {
                  "data": "外呼进度",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h2",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "ii-ui-statisticssubcard-content",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "ii-ui-statisticssubcard-content-value statisticssubcard-value",
              },
              "children": Array [
                Node {
                  "data": "0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "ii-ui-statisticssubcard-content-unit statisticssubcard-unit",
                },
                "children": Array [
                  Node {
                    "data": "%",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Node {
              "attribs": Object {
                "class": "ii-ui-statisticssubcard-content-unit statisticssubcard-unit",
              },
              "children": Array [
                Node {
                  "data": "%",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "ii-ui-statisticssubcard-content-value statisticssubcard-value",
                },
                "children": Array [
                  Node {
                    "data": "0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "ii-ui-statisticssubcard-content",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "ii-ui-statisticssubcard-content-value statisticssubcard-value",
            },
            "children": Array [
              Node {
                "data": "0",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "ii-ui-statisticssubcard-content-unit statisticssubcard-unit",
              },
              "children": Array [
                Node {
                  "data": "%",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "ii-ui-statisticssubcard-content-unit statisticssubcard-unit",
            },
            "children": Array [
              Node {
                "data": "%",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "ii-ui-statisticssubcard-content-value statisticssubcard-value",
              },
              "children": Array [
                Node {
                  "data": "0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "ii-ui-statisticssubcard-header",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "ii-ui-statisticssubcard-header-title statisticssubcard-title",
              },
              "children": Array [
                Node {
                  "data": "外呼进度",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h2",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "ii-ui-statisticssubcard-header-title statisticssubcard-title",
                },
                "children": Array [
                  Node {
                    "data": "外呼进度",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
